v2/devices/FP4.yml

```diff
@@ -0,0 +1,190 @@
+name: "Fairphone 4"
+codename: "FP4"
+formfactor: "phone"
+aliases: ["fp4"]
+doppelgangers: []
+user_actions:
+  confirm_model:
+    title: "Confirm your model"
+    description: "Please check that your device is a Fairphone 4 (FP4)."
+  confirm_os:
+    title: "Confirm OS version"
+    description: "Your device must be running the Fairphone OS version of Android 11 before installing Ubuntu Touch. With a previously installed custom ROM, it won't work!"
+    link: "https://support.fairphone.com/hc/en-us/articles/4405858261777"
+  unlock_phone:
+    title: "Unlock the bootloader"
+    description: "Before installing Ubuntu Touch you must unlock the bootloader of your phone manually. Follow the steps in the linked page if you haven't already."
+    link: "https://support.fairphone.com/hc/en-us/articles/4405858258961-FP4-Manage-the-bootloader"
+  bootloader:
+    title: "Reboot to Bootloader"
+    description: "With the device powered off, press and hold the VOLUME DOWN button and plug the device into your PC via USB. After some seconds you'll see the fastboot mode."
+    image: "phone_power_down"
+    button: true
+  bootloader_from_fastbootd:
+    title: "Reboot to Bootloader"
+    description: 'With the device still at the fastbootd screen, use the volume buttons to select "Reboot to bootloader".'
+    image: "phone_power_down"
+    button: true
+  recovery:
+    title: "Reboot to Recovery"
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection.'
+    image: "phone_power_up"
+    button: true
+  fastbootd:
+    title: "Reboot to fastbootd mode"
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection. Then select "Fastboot mode" from the list.'
+    image: "phone_power_up"
+    button: true
+unlock:
+  - "confirm_model"
+  - "confirm_os"
+  - "unlock_phone"
+# FIXME bootloader_locked? https://www.fairphone.com/en/bootloader-unlocking-code-for-fairphone-3
```

Comment:
The `fastboot:oem_unlock` function allows providing a URL but 1. the code entry page with that is completely misleading on FP4; 2. FP4 would use `fastboot:flashing_unlock` and that doesn't use an URL

I already put "Unlock the bootloader" into the manual confirmation before flashing and expect users to follow that, I guess that's enough?

Reply:
`fastboot:assert_var` could be used to make it more user friendly and actually check if the bootloader is unlocked? Or at least the error will be nicer (`Assertion error: expected bootloader variable unlocked to be yes but got no`)

Reply:
Sounds like a good idea, will try to implement this next week

Reply:
@maciek134 I guess it looks well enough like this?

![image](https://user-images.githubusercontent.com/3768500/183677357-356701af-4870-4650-a6c2-94e71d6069a5.png)


Doesn't show the `link` though that's configured in `unlock_phone`?

```yml
      - actions:
          - fastboot:assert_var:
              variable: "unlocked"
              value: "yes"
        fallback:
          - core:user_action:
              action: "unlock_phone"
```

Reply:
Fixed with https://github.com/ubports/ubports-installer/pull/2798 but I don't consider this to be a blocker for the config.